autospec/tarball.py

```diff
@@ -573,18 +651,22 @@ def process(url_arg, name_arg, ver_arg, target, archives_arg, filemanager):
     version = ver_arg
     archives = archives_arg
     tarfile = os.path.basename(url_arg)
-    # determine name and version of package
-    name_and_version(name_arg, ver_arg, filemanager)
     # determine build pattern and build requirements from url
     detect_build_from_url(url)
+    # Create the download path for content and set build.download_path
+    create_download_path(target)
+    # determine name and version of package
+    name, rawname, version = name_and_version(name_arg, ver_arg, filemanager)
+    # '-' aren't allowed in the version string for spec files
+    # rpmbuild will fail so replace them with '.'
+    if '-' in version:
+        version = version.replace('-', '.')
     # set gcov file information, must be done after name is set since the gcov
     # name is created by adding ".gcov" to the package name (if a gcov file
     # exists)
     set_gcov()
     # download the tarball to tar_path
-    tar_path = download_tarball(target)
-    # write the sha of the upstream tarfile to the "upstream" file
-    write_upstream(get_sha1sum(tar_path), tarfile)
+    tar_path = check_or_get_file(url, os.path.basename(url))
```

Comment:
I think the second argument here can be `tarfile`, correct?

Reply:
Indeed, fixing.